GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1304,6 +1310,21 @@ private void UpdatePlaceholders()
 
                     if (keepFolder)
                     {
+                        if (this.TryGetFolderDataFromTreeUsingPath(folderPlaceholder.Path, out FolderData folderData))

```

Comment:
What do you think about guarding this section with a config setting? Of the two big internal monorepos, one needs this for correctness, the other does not. I worry that the cost here might be pretty large for no real gain if the dev doesn't need BuildXL. Thoughts?

Reply:
I am going to first see what performance on the windows repo looks like and how much this is going to add but this was definitely on my mind as I was writing this code.